apps/client/src/app/modules/list/data/ListRowSerializationHelper.ts

```diff
@@ -0,0 +1,315 @@
+    const retval = { ...instance }
+    if (instance != null && !instance.name)
+      retval.name = this.getNameIfExists(this.l12n.getInstanceName(instance.id))
+    return retval;
+  }
+
+  private serializeInstances(instances: Instance[]): InstanceWithName[] {
+    return instances && instances.length > 0 ? instances.map((r: Instance) => this.applyIntanceName(r)) : undefined;
+  }
+
+  private serializeReducedFrom(reducedFrom: ItemNameHolder[]): ItemNameHolder[] {
+    return reducedFrom && reducedFrom.length > 0 ? reducedFrom.map((r: any) => this.applyItemName(r)) : undefined;
+  }
+
+  private serializeTreasures(treasures: ItemNameHolder[]): ItemNameHolder[] {
+    return treasures && treasures.length > 0 ? treasures.map((r: any) => this.applyItemName(r)) : undefined;
+  }
+
+  private serializeMasterbooks(masterbooks: ItemNameHolder[]): ItemNameHolder[] {
+    return masterbooks && masterbooks.length > 0 ? masterbooks.map((r: any) => this.applyItemName(r)) : undefined;
+  }
+
+  private serializeGardening(gardening: any) {
+    if (!Number.isInteger(gardening))
+      return undefined;
+    return {
+      ...this.applyItemName({ id: gardening }),
+      url: `http://ffxivgardening.com/seed-details.php?SeedID=${gardening}`,
+    };
+  }
+
+  private serializeRequires(item: ListRow): ItemNameHolder[] {
+    return item.requires ? item.requires.map((r: Ingredient) => this.applyItemName(r)) : undefined;
+  }
+
+  private serializeGathering(gathering: GatheredBy): GatherWithName {
+    const retval = gathering ? {
+      ...gathering,
+      //name: this.getJobAbbreviationFromId(LayoutOrderService.getJobIdFromGather(gathering.type)),
+      //unclear what Id to use/convert to... 
+      name: gathering.icon ? gathering.icon.substr(gathering.icon.lastIndexOf("/"), 3) : undefined,
+      nodes: gathering.nodes ? gathering.nodes.map((n: StoredNode) => this.serializeGatheringNode(n)) : []
+    } : undefined;
+    return retval;
+  }
+
+  private serializeGatheringNode(n: StoredNode): any {
+    return {
+      ...n,
+      limitType: n.limitType ? this.i18nTools.getName(n.limitType) : undefined,
+      zoneName: this.getZoneName(n),
+      coords: n.coords ? { x: n.coords[0], y: n.coords[1] } : undefined
+    };
+  }
+
+  public getJsonExport(dc: string, server: string, rows: ListRow[], finalItems: ListRow[]): any {
+    //convert all the desynth data into a single key value pair/dictionary (effectively dedupes for large dumps)
+    const desynthMap = {}
+    const desynthData = rows
+      .map((item: ListRow) => {
+        const desynths = ListRowSerializationHelper.getData(item, DataType.DESYNTHS);
+        return desynths && desynths.length > 0 ? desynths.map((r: any) => this.applyItemName({ id: r })) : undefined
+      })
+      .filter(m => !!m)
+      .reduce((pn, u) => [...pn, ...u], []);
+    for (let i = 0; i < desynthData.length; i++) {
+      desynthMap[desynthData[i].id] = desynthData[i].name;
+    }
+    const retval = {
```

Comment:
I don't understand the purpose of `retval = ...; return retval;`, why not return the value directly?